Add QwQ model

darraghdog opened the pull request:
@Kipok I am running through the tutorial and saw the QwQ model was not download/converted. If this was intentional you can close the PR.

shtoshni commented:
@darraghdog - We can use hf models directly with trtllm, and the server automatically downloads the model now - https://nvidia.github.io/TensorRT-LLM/quick-start-guide.html#run-offline-inference-with-llm-api

darraghdog commented:
Thanks, I think I need to update the trtllm container for that - I'm using `0.21.0` now. 

In the [tutorial](https://github.com/NVIDIA/NeMo-Skills/blob/main/docs/tutorials/posts/omr-simple-recipe.md ) which I'm following, it uses `model=/workspace/QwQ-32B` as input. If this was changed to `model="Qwen/QwQ-32B"` would it automatically download and convert it ? 

```
generate(
    ctx=wrap_arguments(
        f"++prompt_config=generic/math "
        f"++inference.temperature=0.6 "
        f"++inference.tokens_to_generate=8192 "
    ),
    cluster=cluster,
    input_file="/workspace/sdg/extracted-problems.jsonl",
    output_dir="/workspace/sdg/solutions",
    expname="solution-generation",
    run_after="problem-extraction",
    model="/workspace/QwQ-32B",
    server_type="trtllm",
    server_gpus=num_gpus,
    # remove these parameters to disable wandb logging
    log_samples=True,
    wandb_group="sdg",
)```

shtoshni commented:
@darraghdog - You can just pass `model="Qwen/QwQ-32B"` 
The model will be downloaded to `HF_HOME` if it's not already downloaded there.
And no need for any model conversion as well.

shtoshni commented:
@Kipok - Do you think we should remove references from old tutorials/releases about explicitly downloading the model?

Kipok commented:
@darraghdog can you please try to just use `Qwen/QwQ-32B` and change to that if it works for you? If not, please let us know

darraghdog commented:
I updated to your recommendation @shtoshni .. Most of the the step are running through well - it makes it simpler. Will let you know when I'm through it all. 

It seems below like a pre-download & mount is required for the `run_sdg.py` QwQ step when using trtllm; but it works fine with vllm, so for now I switched to in the PR.  
 
```
Traceback (most recent call last):
  File "/Users/dhanley/Documents/ns_tutorials/run_sdg.py", line 31, in <module>
    generate(
  File "/opt/anaconda3/lib/python3.11/site-packages/nemo_skills/pipeline/app.py", line 51, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/nemo_skills/pipeline/generate.py", line 279, in generate
    new_task = pipeline_utils.add_task(
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/nemo_skills/pipeline/utils/exp.py", line 327, in add_task
    server_cmd, num_server_tasks = get_server_command(**server_config, cluster_config=cluster_config)
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/nemo_skills/pipeline/utils/server.py", line 204, in get_server_command
    check_if_mounted(cluster_config, model_path)
  File "/opt/anaconda3/lib/python3.11/site-packages/nemo_skills/pipeline/utils/mounts.py", line 54, in check_if_mounted
    raise ValueError(f"The path '{path_to_check}' is not mounted. Check cluster config.")
ValueError: The path 'Qwen/QwQ-32B' is not mounted. Check cluster config.
```

shtoshni commented:
@darraghdog - It seems like you're not on the latest main because this should not be happening on the latest main - See here - https://github.com/NVIDIA/NeMo-Skills/blob/main/nemo_skills/pipeline/utils/server.py#L126

shtoshni commented:
@darraghdog - Fixed the docs thanks to your observation!

darraghdog commented:
Closing this PR as changes are in @shtoshni  PR; thanks!
